### ISSUE TITLE
Node 20: `ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time`

### ISSUE THREAD

**Issue by dobladov** [cypress-io/cypress#26583]
### Test code to reproduce

Adding `"type": "module"` to `package.json` creates the error, if it's not present and uses `require`, _cypress_ works as expected.

To recreate the issue run the following commands with node `20`
```sh
npm install
npm run test:e2e
```
This error should be present in the console
```
(Use `node --trace-warnings ...` to show where the warning was created)
(node:9238) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:9238) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:9238) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:9238) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:9238) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:9238) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
Using node 19 has shows the error but is less frequent.
```

Using node `19` shows the error, but it's less frequent.

Minimum repository to re-create the issue https://github.com/dobladov/cypress-module-error

### Cypress Mode

both modes

### Cypress Version

12.10.0

### Browser Version

Chrome latest

### Node version

20.0.0

### Operating System

macOs Ventura 13.3

### Memory Debug Logs

```shell
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13446) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
```

**Comment by mschile** [comment]
Hi @dobladov 👋, thanks for logging this issue! I was able to reproduce it using your repository. It appears the Cypress tests themselves still run correctly. I am going to send your issue onto the team so we can investigate further.

**Comment by dobladov** [comment]
The issue is solved with node [v20.1.0](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V20.md#2023-05-03-version-2010-current-targos).

The message still appears once, but it's not constantly being logged until it goes out of memory.

**Comment by kaiyoma** [comment]
I am seeing this issue with Cypress 11 and Node 18, also in an ESM environment.
